pure-selenium/src/main/java/org/example/pages/InventoryPage.java

```diff
@@ -49,17 +48,15 @@ public java.util.List<Double> getDisplayedPrices() {
                 .toList();
     }
 
-
-    public void clickCartIcon() {
+    public CartPage clickCartIcon() {
         cartIcon.click();
+        return new CartPage(driver);
     }
 
-    public void addBikeLightToCart() {
-        addBikeLightToCartButton.click();
-    }
-
-    public void addBackpackToCart() {
-        addToCartButton.click();
+    public void addProductToCartByName(String productName) {
```

Comment:
Use component-based approach for clarity and maintainability, below you can find info how to create component
https://www.selenium.dev/documentation/test_practices/encouraged/page_object_models/